lib/query/proposal/search_dsl.rb

```diff
@@ -28,19 +28,39 @@ def apply_authz?
       end
 
       def composite_query_string
-        if client_query && query_str
-          "(#{query_str}) AND (#{client_query.map { |k, v| "#{k}:(#{v})" }.join(' ')})"
-        elsif client_query
-          client_query.map { |k, v| "#{k}:(#{v})" }.join(" ")
-        elsif query_str
-          query_str
+        if client_query.present? && query_str.present?
+          "(#{query_str}) AND (#{client_query})"
+        elsif client_query.present?
+          client_query.to_s
+        elsif query_str.present?
+          query_str.to_s
+        else
+          ""
         end
       end
 
+      def client_query
+        fielded = params[current_user.client_model_slug.to_sym]
+        munge_fielded_params(fielded) if fielded
+        FieldedSearch.new(fielded)
+      end
+
       private
 
-      def client_query
-        params[client_data_type.underscore.tr("/", "_").to_sym]
+      def munge_fielded_params(fielded)
+        if fielded[:created_at].present? && fielded[:created_within].present?
+          munge_created_at_field(fielded)
+        end
+        # do not calculate more than once, or when created_at is null
+        fielded.delete(:created_within)
+      end
+
+      def munge_created_at_field(fielded)
```

Comment:
from the hackers dictionary, "munge" is:
> (1) a verb, used in a derogatory sense, meaning to imperfectly transform information

in this case, it is re-writing the user-supplied string value of `created_at` with a stringified range query, based on the presence and parse-ability of the `created_within` field.

If "munge" doesn't work for you, what word would you prefer?